core/editor/src/document/document_message_handler.rs

```diff
@@ -638,6 +641,47 @@ impl MessageHandler<DocumentMessage, &InputPreprocessor> for DocumentMessageHand
 					);
 				}
 			}
+			AlignSelectedLayers(dimension, aggregate) => {
+				let selected_paths = self.selected_layers_sorted();
+				if selected_paths.len() == 0 {
+					return;
+				}
+
+				let selected_layers = selected_paths.iter().map(|path| {
+					let layer = self.active_document().document.layer(path).unwrap();
+					let point = {
+						let bounding_box = layer.bounding_box(layer.transform, layer.style).unwrap();
+						match aggregate {
+							AlignAggregate::Min => bounding_box[0],
+							AlignAggregate::Max => bounding_box[1],
+							AlignAggregate::Average => bounding_box[0].lerp(bounding_box[1], 0.5),
```

Comment:
Average means it's adding up all the center coordinates of all the shapes then taking the mean value and setting the centers of each to that mean value, which is useful but is not the normal idea of "Center" in graphics software. "Average" will be another button [accessible in popover](https://files.keavon.com/-/BulkyAltruisticMonarch/capture.png) to the right of these alignment buttons. So please keep the `Average` case but add a `Center` case above it which centers everything in the center of the union bounding box of all shape bounding boxes. So you do what you've done here with the `Min` and `Max` case by using `reduce()` to find the min and max of all the bounding boxes, then you take the center (sum divided by two) of the min and max values.